lang/src/org/partiql/lang/eval/visitors/AggregationVisitorTransform.kt

```diff
@@ -0,0 +1,364 @@
+                    projectExpr(id(key.uniqueAlias, caseSensitive(), unqualified()), key.publicAlias)
+                }.toMutableList()
+
+                contextStack.last().groupAsAlias?.let { alias ->
+                    val item = projectExpr(id(alias, caseSensitive(), unqualified()), alias)
+                    projectionItems.add(item)
+                }
+                projectList(projectionItems)
+            }
+        }
+        return super.transformProjectionProjectStar(node)
+    }
+
+    override fun transformProjectionProjectValue(node: PartiqlAst.Projection.ProjectValue): PartiqlAst.Projection =
+        PartiqlAst.build {
+            projectValue(
+                value = itemTransform.transformExpr(node.value),
+                metas = node.metas
+            )
+        }
+
+    override fun transformSortSpec_expr(node: PartiqlAst.SortSpec) = itemTransform.transformSortSpec_expr(node)
+
+    override fun transformExprSelect_having(node: PartiqlAst.Expr.Select): PartiqlAst.Expr? = node.having?.let { having ->
+        itemTransform.transformExpr(having)
+    }
+
+    override fun transformProjectionProjectList(node: PartiqlAst.Projection.ProjectList): PartiqlAst.Projection =
+        PartiqlAst.build {
+            projectList(
+                projectItems = node.projectItems.map { item ->
+                    when (item) {
+                        is PartiqlAst.ProjectItem.ProjectExpr -> {
+                            val projectionAlias = item.asAlias ?: throw SemanticException(
+                                err = Problem(
+                                    item.metas.sourceLocationMeta ?: UNKNOWN_SOURCE_LOCATION,
+                                    details = SemanticProblemDetails.MissingAlias
+                                )
+                            )
+
+                            projectExpr_(
+                                expr = itemTransform.transformExpr(item.expr),
+                                asAlias = projectionAlias
+                            )
+                        }
+                        else -> item
+                    }
+                },
+                metas = node.metas
+            )
+        }
+
+    /**
+     * Recursively searches through a [PartiqlAst.Projection] to find [PartiqlAst.Expr.CallAgg]'s, but does NOT recurse
+     * into [PartiqlAst.Expr.Select]. Designed to be called directly using [containsAggregations].
+     */
+    private object AggregationFinder : PartiqlAst.Visitor() {
+        var hasAggregations: Boolean = false
+
+        fun containsAggregations(node: PartiqlAst.Projection): Boolean {
+            this.walkProjection(node)
+            return this.hasAggregations.also { this.hasAggregations = false }
+        }
+
+        override fun visitExprCallAgg(node: PartiqlAst.Expr.CallAgg) {
+            hasAggregations = true
+        }
+
+        override fun walkExprSelect(node: PartiqlAst.Expr.Select) { }
```

Comment:
May be worth to leave a comment here to say this is intentionally left empty because we do not want to go in nested select. Or add a return here?